include/zephyr/arch/arm/asm_inline_gcc.h

```diff
@@ -105,6 +105,25 @@ static ALWAYS_INLINE bool arch_irq_unlocked(unsigned int key)
 	return key == 0U;
 }
 
+#ifdef CONFIG_ZERO_LATENCY_IRQS
+
+static ALWAYS_INLINE unsigned int arch_zli_lock(void)
+{
+	unsigned int key;
+
+	key = __get_PRIMASK();
+	__disable_irq();
```

Comment:
Does this need a call to __ISRB()? I ask because not only do I not know, but I observe that the arch_irq_lock() implementation for ARMV7_M_ARMV8_M_MAINLINE has it and ZERO_LATENCY_IRQS depends on CPU_CORTEX_M_HAS_BASEPRI which in turn depends on ARMV7_M_ARMV8_M_MAINLINE.